pyriemann/regression.py

```diff
@@ -0,0 +1,174 @@
+        Metric for kernel matrix computation.
+    Cref : None | ndarray, shape (n_channels, n_channels)
+        Reference point for kernel matrix computation. If None, the mean of
+        the training data according to the metric is used.
+    **kwargs
+        Keyword arguments passed to svc_func.
+
+    Attributes
+    ----------
+    svc_ : svc_func instance
+        Fitted SVC with precomputed Riemannian kernel matrix.
+    data_ : ndarray, shape (n_matrices, n_channels, n_channels)
+        Training data.
+
+    Notes
+    -----
+    .. versionadded:: 0.2.8
+
+    References
+    ----------
+    .. [1] A. Barachant, S. Bonnet, M. Congedo, and C. Jutten.
+        Classification of covariance matrices using a Riemannian-based kernel
+        for BCI applications". In: Neurocomputing 112 (July 2013), pp. 172-178.
+    """
+
+    def __init__(self, metric='riemann', Cref=None,
+                 **kwargs):
+        """Init."""
+        self.svr_func = sklearnSVR
+        self.Cref = Cref
+        self.metric = metric
+        self.svr_params = kwargs
+        self.svr_ = self.svr_func(kernel='precomputed', **self.svr_params)
+
+    def __setattr__(self, name, value):
+        """Enable setting attributes for SVC subclass."""
+        if 'svr_' in self.__dict__.keys():
+            if name in self.svr_.__dict__.keys():
+                self.svr_.set_params(**{name: value})
+                return
+        super().__setattr__(name, value)
+
+    def fit(self, X, y):
+        """Fit.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+        y : ndarray, shape (n_matrices, 1)
+            labels corresponding to each matrix.
+
+        Returns
+        -------
+        self : Riemannian SVC instance
+            The SVC instance.
+        """
+        kernelmat = kernel(X, Cref=self.Cref, metric=self.metric)
+        self.data_ = X
+        self.svr_ = self.svr_.fit(kernelmat, y)
+
+        return self
+
+    def predict(self, X):
+        """Get the predictions.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_matrices, n_channels, n_channels)
+            Set of SPD matrices.
+
+        Returns
+        -------
+        pred : ndarray of int, shape (n_matrices, 1)
+            Predictions for each matrix according to the SVC.
+        """
+        test_kernel_mat = kernel(X,
+                                 self.data_,
+                                 Cref=self.Cref,
+                                 metric=self.metric)
+        return self.svr_.predict(test_kernel_mat)
+
+
+class KNNRegression(MDM):
+    """Regression by K-Nearest-Neighbors.
+
+    Reression by nearest Neighbors. For each point of the test set, the
+    pairwise distance to each element of the training set is estimated. The
+    value is calculated according to the softmax average w.r.t. distance of
+    the k nearest neighbors.
```

Comment:
Can you add a reference for this algorithm?

Reply:
Uhm, I didn't base it on any reference.

Reply:
It might be entirely stupid.

Reply:
Done.